core/src/main/java/hudson/model/ItemGroup.java

```diff
@@ -55,6 +58,17 @@
      */
     Collection<T> getItems();
 
+    /**
+     * Gets all the items in this collection in a read-only view
+     * that matches supplied Predicate
+     * @since TODO
+     */
+     default Collection<T> getItems(Predicate<T> pred) {
+         return getItems().stream()
```

Comment:
Would it make sense to create a `default Stream<T> getItemsStream()` method so that ItemGroup implementations can supply more efficient stream implementations which would not require creating a full list first. Will be important if we ever do job lazy loading and so on

Reply:
It probably would be 2 functions
`default Stream<T> getItemsStream()`
and
`default Stream<T> getItemsStream(Predicate<T> pred)`
So it still does preauth filtering since once ItemGroup is responsible for security checks. Unless we want to offload that to the caller

Reply:
Added stream methods.